server/Controllers/changePassword.js

```diff
@@ -0,0 +1,45 @@
+import bcrypt from "bcrypt";
+import UserModel from "../Schemas/User";
+
+const validatePassword = (password) => {
+  const reg = /(?=^.{8,}$)((?!.*\s)(?=.*[A-Z])(?=.*[a-z]))((?=(.*\d){1,})|(?=(.*\W){1,}))^.*$/;
+  return reg.test(password);
+};
+
+const changePassword = async (req, res) => {
+  const encodedNewPassword = bcrypt.hashSync(req.body.newPassword, 10);
+
+  if (!(await validatePassword(req.body.newPassword))) {
+    res.status(400).send();
+  } else {
+    try {
+      const userInfos = await UserModel.findById(req.userId);
+      if (userInfos === null) {
+        res.status(404).send();
+      } else if (
+        !(await bcrypt.compareSync(req.body.oldPassword, userInfos.password))
+      ) {
+        res.status(401).send();
+      } else {
+        try {
```

Comment:
In fact one try/catch work for all the code. You're handling the catch part the same way so I think we only need one